workflow/rules/trim_3prime.smk

```diff
@@ -0,0 +1,88 @@
+if is_3prime_experiment:
+
+    # Rule cutadapt1 checks and removes poly-A tails and sequence qualilty score <20.
+    # reasoning behind parameters:
+    #   * `-m 20`:
+    #   * Discards any read under 20bp by -m 20
+    #   * `-O 20 -a ""polyA=A{20}"" -a ""QUALITY=G{20}"" -n 2`:
+    #   * Removes reads having polyA (18bp) or polyG (18bp) and repeat the process twice by -n 2
+    # In short, it removes reads that are predominantly polyA or polyG.
+
+    # Rule cutadapt2 checks if reads are obtained from nextseq and removes adapters specific to nextseq
+    # reasoning behind parameters:
+    #   * `-m 20`:
+    #   * Discards any read under 20bp by -m 20
+    #   * `--nextseq-trim=10`
+    #   * Quality score<20 is trimmed, based on NextSeq/NovaSeq 2-color SBS system
+    #   * `-a A{18}AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC;min_overlap=3;max_error_rate=0.100000`
```

Comment:
In order to make it clear that the current implementation just works for this specific lexogene setup, we should add a key to the config:
```
experiment:
  3-prime-rna-seq:
    # Specify vendor of the used protocol. Currently, only lexogene is supported.
    vendor: lexogene
```
And check for this value in this section of the code, raising and error if the vendor is not defined as lexogene.

Reply:
Noted. Updated the code. Just as a note the name of the vendor is `lexogen` instead of `lexogene`.

Reply:
This also has to be reflected in the schema.

Reply:
Done.